File: project/src/main/java/awale/game/Bot2.java
package awale.game;

import awale.action.Action;
import awale.ai.Evaluation1;
import awale.ai.Evaluation2;

public class Bot2 extends Bot {

	public Bot2() {
		super();
		setName("Bot2");
	}

	@Override
	public Action chooseAction() {
		int depth;
		int mobility = mobility(getBoard());

		if (mobility < 9)
			depth = 8;
		else if (mobility < 20)
			depth = 7;
		else
			depth = 6;

		if (getNbSeed() + getOpponent().getNbSeed() > 5)
			minMax.setEvaluation(new Evaluation2());
		else
			minMax.setEvaluation(new Evaluation1());

		Action action = minMax.decisionAlphaBeta(getBoard(), depth, true);
		System.out.println(getName() + " play " + action);

		return action;
	}
}

File: projectC/awale/src/game/bot2.c
#include "../../headers/game/bot2.h"

Bot2 createBot2()
{
    Bot2 bot2;
    bot2.bot = createBot("Bot2");
    // printf("ttt %s\n", bot2.bot->player->name);
    // bot2.bot->chooseAction = chooseAction_bot2;

    return bot2;
}

Action chooseAction_bot2(Bot *bot)
{
    // int depth = 6;
    // int mob = mobility(bot->player, bot->player->board);

    // if (mob < 9)
    // {
    //     depth = 8;
    // }
    // else if (mob < 20)
    // {
    //     depth = 7;
    // }

    // Evaluation evaluation;

    // if (*(bot->player->nbSeed) + *(bot->player->opponent->nbSeed) > 5)
    // {

    //     evaluation.evaluate = evaluate_evaluation2;
    // }
    // else
    // {
    //     evaluation.evaluate = evaluate_evaluation1;
    // }

    // setEvaluation(&(bot->minMax), &(evaluation));

    // Action action = decisionAlphaBeta(&(bot->minMax), bot->player->board, depth, true);
    // printf("%s play %s", bot->player->name, toStringAction(&(action)));
    Action action;

    return action;
}
